components/system/components/GlobalCarousel.js

```diff
@@ -0,0 +1,130 @@
+import { css } from "@emotion/react";
+import * as React from "react";
+import * as Constants from "~/common/constants";
+import * as SVG from "~/components/system/svg";
+
+const STYLES_BACKGROUND = css`
+  position: fixed;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: ${Constants.system.pitchBlack};
+  color: ${Constants.system.white};
+  z-index: ${Constants.zindex.modal};
+`;
+
+const STYLES_CLOSE_ICON = css`
+  height: 24px;
+  position: absolute;
+  top: 8px;
+  right: 8px;
+  cursor: pointer;
+`;
+
+const STYLES_PREVIOUS_ICON = css`
+  height: 24px;
+  position: absolute;
+  top: 50%;
+  left: 8px;
+  transform: translateY(-50%);
+  cursor: pointer;
+`;
+
+const STYLES_NEXT_ICON = css`
+  height: 24px;
+  position: absolute;
+  top: 50%;
+  right: 8px;
+  transform: translateY(-50%);
+  cursor: pointer;
+`;
+
+export class GlobalCarousel extends React.Component {
+  state = {
+    slides: null,
+    currentSlide: 0,
+  };
+
+  componentDidMount = () => {
+    window.addEventListener("create-carousel", this._handleCreate);
+    window.addEventListener("delete-carousel", this._handleDelete);
+
+    // Handle Key downs
```

Comment:
Not a big deal, for the future lets keep comment structure with name identifiers

```
// NOTE(aminejvm):
// Handle key downs
```